jax-stuff.md

```diff
@@ -80,27 +80,20 @@ _styles: >
 
 ## How Does Parallelism Work in JAX?
 
-JAX supports three schools of thought for multi-device programming:
+JAX supports two schools of thought for multi-device programming:
 
-1. **Compiler, take the wheel!** Let the XLA compiler automatically partition arrays and decide what communication to add to facilitate a given program. This lets you take a program that runs on a single device and automatically run it on thousands without changing anything.
-2. **JAX, take the wheel!** Automatic parallelism is great, but sometimes the compiler does something crazy. Explicit sharding lets you write single-device code like usual, but have JAX handle sharding propagation (not the compiler). This means JAX can ask you for clarification when it's unclear what you want.
-3. **Just let me write what I mean, damnit!** While compilers are nice, they sometimes do the wrong thing and add communication you don't intend. Sometimes we want to be explicit about exactly what communication you intend to run.
+1. **Compiler, take the wheel!** Let the compiler automatically partition arrays and decide what communication to add to facilitate a given program. This lets you take a program that runs on a single device and automatically run it on thousands without changing anything.
+2. **Just let me write what I mean, damnit!** While compilers are nice, they sometimes do the wrong thing and add communication you don't intend. Sometimes we want to be explicit about exactly what communication you intend to run.
```

Comment:
I think we should only mention Explicit and Manual mode here if you only want 2 sections.

Mentioning Compiler and Manual doesn't seem right to me. 

This section in JAX docs: https://docs.jax.dev/en/latest/notebooks/explicit-sharding.html#using-a-mixture-of-sharding-modes does a pretty good job at explaining all modes. Maybe we should copy it here as I originally did in the rewrite :)

Reply:
I disagree. It is an important distinction since it is a different way of programming and it affects how the code will be written.

> That feels like the important thing.

That was the old way of thinking about it when there were only 2 modes. We have 3 modes right now. So the working needs to evolve.